File: Carly_Michael/jectacular-ql/src/app/domain/ast/expressions/binary-expression.ts
import {BinaryOperator, Expression} from './expression';
import {Variable} from './variable';
import {Location} from '../location';
import * as _ from 'lodash';

export abstract class BinaryExpression extends Expression {

  constructor(public left: Expression, public right: Expression, public operator: BinaryOperator, location: Location) {
    super(location);
  }

  getVariables(): Variable[] {
    return _.flatten(this.left.getVariables().concat(this.right.getVariables()));
  }
}

File: Rasha_Toine/Rasha/QLForm/src/ql/ast/expression/BinaryExpression.java
package ql.ast.expression;

public abstract class BinaryExpression extends Expression {
	private Expression left;
	private Expression right;
	
	public BinaryExpression(Expression left, Expression right) {
		this.setLeft(left);
		this.setRight(right);
	}

	public Expression getLeft() {
		return left;
	}

	public void setLeft(Expression left) {
		this.left = left;
	}

	public Expression getRight() {
		return right;
	}

	public void setRight(Expression right) {
		this.right = right;
	}
}
